flight/Modules/VtolPathFollower/vtol_follower_fsm.c

```diff
@@ -233,16 +240,56 @@ const static struct vtol_fsm_transition fsm_land_home[FSM_STATE_NUM_STATES] = {
 
 };
 
-//! Tracks how many times the fsm_static has been called
-static uint32_t current_count, set_time_count, timer_duration = 0;
+//! Specifies a time since startup in ms that the timeout fires.
+static uint32_t vtol_fsm_timer_expiration = 0;
 
-void configure_timeout(int32_t s)
+/**
+ * Sets up an interval timer that can be later checked for expiration.
+ * @param[in] ms Relative time in millseconds.  0 to cancel pending timer, if any
+ */
+static void vtol_fsm_timer_set(int32_t ms)
 {
-	set_time_count = current_count;
-	timer_duration = s;
+	if (ms) {
+		uint32_t now = PIOS_Thread_Systime();
+		vtol_fsm_timer_expiration = ms+now;
+
+		// If we wrap and get very unlucky, make sure we still
+		// have a timer 1ms later.
+		if (vtol_fsm_timer_expiration == 0) {
```

Comment:
Good catch! That one in 4294967296 times this failed would be very hard to reproduce but would essentially leave a quad hanging in RTH.